libraries/js/text/regex.effekt

```diff
@@ -1,33 +1,242 @@
 module text/regex
 
 import string
+import option
+import list
+
 
 extern type Regex
 
-record Match(matched: String, index: Int)
+// matches and groups
+record Range(start: Int, end: Int)
+record Group(matched: String, index: Option[Range])
+record Match(matched: String, start: Int, end: Int,  groups: List[Group])
+
+/**
+ * Specifies flags that modify regular expression behavior.
+ *
+ * Flags:
+ * - Global: Matches all occurrences in the input string.
+ * - IgnoreCase: Performs case-insensitive matching.
+ * - Multiline: Treats ^ and $ as start/end of each line.
+ * - GenerateIndices: Enables capture group index retrieval. This flag is ignored
+ *   when using Node.js versions below 16.0.0.
+ *
+ * Usage:
+ * ```effekt
+ * val regex = regex("pattern", [Global(), IgnoreCase(), GenerateIndices()])
+ * ```
+ *
+ * Note: GenerateIndices is required for capture group indexing.
+ * Without it, capture group indices will be None.
+ */
+type RegexFlags {
+  Global();
+  IgnoreCase();
+  Multiline();
+  GenerateIndices();
+}
```

Comment:
Out of curiosity: 

1. is it possible to implement global matching in terms of local one (starting from a given index) _within_ Effekt?
2. is it possible to implement multiline matching _within_ Effekt by preprocessing the regular expression string? 
3. Can we always require `GenerateIndices` and drop it here?

Reply:
1. Is certainly possible and would possible allow for shared logic across different backend. 
2. multiline should also be possible to implement within Effekt even though I'm not sure how that would impact performance. In my understanding most implementations use simd acceleration for the line ending search or embed a newline stopping criteria directly in the regex engine. 
3. Yes we can.